### PR TITLE
Blog App: Add Devise

### PR DESCRIPTION
**Pull request by anyars-encarta**
In this project, I created functionalities to meet the requirements below:

- [ ] Remove current_user method in ApplicationController, devise will provide us with one.

- [ ] [Install and setup devise](https://github.com/heartcombo/devise#getting-started).

- [ ] Can register a new user.

- [ ] User logs in with a combination of email and password.

- [ ] Hashed password should be stored in the database.

- [ ] Ask for confirmation of email.

- [ ] Can reset password.

- [ ] [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

**Review by OyePriscilla — COMMENTED** [review]
Hi @anyars-encarta 👏 👏, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a pair-programming project, that does not have contribution from your partners. Kindly open a new PR with contributions from your coding partners. 🙏 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+  # attr_reader :current_user
 
-  private
+  # private
 
-  def set_current_user
-    @current_user = User.first
+  # def set_current_user
+  #   @current_user = User.first
+  # end
```
Comment (shahadat3669) [inline app/controllers/application_controller.rb]: - [ ] Well, in multiple files, I noticed that you have commented out code. This is not a good practice as commented-out code can be confusing and clutter the source files. If those lines of code are no longer necessary, please consider removing them completely to keep your codebase clean and maintainable. Thank you for your attention to code quality! 👍🧹📄

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @BilalLiaquat7,  @anyars-encarta, @binyamolango 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights

✅ everyone has contributed
✅ Devise has been installed and setup
✅ A hashed password is stored in the database


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AmaduKamara — APPROVED** [review]
# Status: APPROVED 🟢 🟢 🟢

Hi @anyars-encarta,

Good job working on the changes highlighted by the previous reviewer 👍🏽

Your project is complete! There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You set up Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Set up devise gem correctly as expected 👍🏽
✅ Registration and Sign in features work fine 👍🏽
✅ Confirmation feature works fine as expected 👍🏽 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**